src/PrestaShopBundle/Controller/Admin/Sell/Catalog/DiscountController.php

```diff
@@ -78,33 +112,105 @@ public function indexAction(
+    }
+
+    #[DemoRestricted(redirectRoute: 'admin_discounts_index')]
+    #[AdminSecurity("is_granted('create', request.get('_legacy_controller'))", redirectRoute: 'admin_discounts_index')]
+    public function editAction(
+        Request $request,
+        #[Autowire(service: 'prestashop.core.form.identifiable_object.builder.discount_form_builder')]
+        FormBuilderInterface $formBuilder,
+        #[Autowire(service: 'prestashop.core.form.identifiable_object.handler.discount_form_handler')]
+        FormHandlerInterface $formHandler,
+        DiscountRepository $discountRepository,
+        int $discountId,
+    ) {
         try {
-            $handlerResult = $formHandler->handle($form);
-            if ($handlerResult->isSubmitted() && $handlerResult->isValid()) {
-                $this->addFlash('success', $this->trans('Successful creation', [], 'Admin.Notifications.Success'));
+            $discount = $discountRepository->get(new DiscountId($discountId));
+        } catch (DiscountNotFoundException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+
+            return $this->redirectToRoute('admin_discounts_index');
+        }
+
+        try {
+            $form = $formBuilder->getFormFor($discountId, [], [
+                'discount_id' => $discountId,
+                'discount_type' => $discount->type,
+                'method' => Request::METHOD_POST,
+            ]);
+        } catch (DiscountNotFoundException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
 
-                // @todo: redirect to edition page when it is implemented
+            return $this->redirectToRoute('admin_discounts_index');
+        }
+
+        try {
+            $form->handleRequest($request);
+            $result = $formHandler->handleFor($discountId, $form);
+
+            if ($result->isSubmitted()) {
+                if ($result->isValid()) {
+                    $this->addFlash('success', $this->trans('Successful update', [], 'Admin.Notifications.Success'));
+
+                    return $this->redirectToRoute('admin_discount_edit', ['discountId' => $discountId]);
+                } else {
+                    // Display root level errors with flash messages
+                    foreach ($form->getErrors() as $error) {
+                        $this->addFlash('error', sprintf(
+                            '%s: %s',
+                            $error->getOrigin()->getName(),
+                            $error->getMessage()
+                        ));
+                    }
+                }
             }
         } catch (Exception $e) {
             $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
         }
 
-        return $this->redirectToRoute('admin_discounts_index');
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Discount/create.html.twig', [
```

Comment:
```suggestion
        return $this->render('@PrestaShop/Admin/Sell/Catalog/Discount/edit.html.twig', [
```
You shsould have a dedicated edit template, even if it will be very similar to the create one